ui/raidboss/data/07-dt/raid/r2s.ts

```diff
@@ -30,6 +32,67 @@ const triggerSet: TriggerSet<Data> = {
     partnersSpreadCounter: 0,
   }),
   triggers: [
+    {
+      id: 'R2S Beat Tracker',
+      type: 'StartsUsing',
+      netRegex: { id: ['9C24', '9C25', '9C26'], capture: true },
+      run: (data, matches) => {
+        if (matches.id === '9C24')
+          data.beat = 1;
+        else if (matches.id === '9C25')
+          data.beat = 2;
+        else
+          data.beat = 3;
+      },
+    },
+    {
+      id: 'R2S Heart Debuff',
```

Comment:
The output for beat1 happens before the Centerstage/Outerstage combo, and is long gone before the towers start appearing.  I'd suggest adding either a `delaySeconds: 17,` or having it fire on the following `Love Me Tender` cast instead.

I noted a handful of other fixes/improvements for r2s, but I'll PR those separately.

Reply:
If someone eats a hit from centerstage/outerstage, they'll get another stack of heart debuff and that would make the delayed callout wrong.

Also the boss sometimes sends out less hearts than the maximum amount to resolve towers, e.g. sometimes you can end up after the towers on a clean run with:

- 3 players with 3 hearts
- 5 players with 2 hearts

So it's possible for someone to read the heart counts and then greed DPS on the boss by soaking an extra hit from center/outerstage.

Reply:
> If someone eats a hit from centerstage/outerstage, they'll get another stack of heart debuff and that would make the delayed callout wrong.

In that case, then wouldn't the original callout be incorrect (e.g. `Soak 0-1 towers`), since if they start with 2 hearts and get hit, they could no longer soak 1 tower?  Maybe a better approach would be to check their heart count at the start of `Love Me Tender` and do the call then, if intentionally failing subsequent mechanics could alter it.

Either way, I found it a bit confusing to get the call so early only to then have it disappear and get a `Centerstage/Outerstage` call, only to then have the original call become relevant again.

Reply:
Yeah, seems like a lose/lose. Maybe it should just be a generic callout, something like "soak towers until you get at least two hearts"? I just have no idea how to word something like that to make sense.

Reply:
> Yeah, seems like a lose/lose. Maybe it should just be a generic callout, something like "soak towers until you get at least two hearts"? I just have no idea how to word something like that to make sense.

TTS players would *love* that 🤣 . 

Yeah, not really any great options here.  I do think you've got me leaning toward just ignoring the starting heart count, since it can change.  So options for a delayed call might be:
1. No call at all -- probably simplest, but feels meh.
2. 'Soak towers (as needed)' -- accurate, don't reference the heart count, players just need to know how it works
3. 'Get to 2 or 3 hearts' -- don't reference the towers, sorta the same thing (i think i prefer option 2 over this though)
4.  'Soak towers - need 2-3 hearts' -- probably closest to what you have, maybe short enough?

I think something like option 4 is probably the best of the bad?  But honestly, I think any is reasonable given the mech.

Reply:
Alright, I think latest commit addresses both of these points? If so, feel free to approve/merge. I'm going to try and sleep before work starts back up again tomorrow.